etcd-client/src/lib.rs

```diff
@@ -220,10 +221,50 @@ mod tests {
             test_kv().await?;
             test_transaction().await?;
             test_lock().await?;
+            test_watch().await?;
             Ok(())
         }))
     }
 
+    async fn test_watch() -> Result<()> {
+        let client = build_etcd_client().await?;
+
+        client
+            .kv()
+            .put(EtcdPutRequest::new("41_foo1", "baz1"))
+            .await?;
+        client
+            .kv()
+            .put(EtcdPutRequest::new("42_foo1", "baz2"))
+            .await?;
+
+        let watch_key = "41_foo1";
+        let mut response = client.watch(KeyRange::key(watch_key)).await;
+        assert_eq!(
+            response
+                .next()
+                .await
+                .unwrap_or_else(|| panic!("Fail to receive reponse from client"))
+                .unwrap_or_else(|e| { panic!("failed to get watch response, the error is: {}", e) })
+                .watch_id(),
+            0
+        );
+
+        client
+            .kv()
+            .put(EtcdPutRequest::new("41_foo1", "baz3"))
+            .await?;
+        let mut watch_response = response
+            .next()
+            .await
+            .unwrap_or_else(|| panic!("Fail to receive reponse from client"))
+            .unwrap_or_else(|e| panic!("failed to get watch response, the error is: {}", e));
+
+        assert_eq!(watch_response.watch_id(), 0);
```

Comment:
assert_eq!(watch_response.watch_id(), 0, "add comments to this assert");

Reply:
Done